hack/fuse-demo/wrap_datamon.sh

```diff
@@ -8,98 +8,232 @@
 setopt ERR_EXIT
 setopt PIPE_FAIL
 
+#####
+
+### util
+
+terminate() {
+    print -- "$*" 1>&2
+    exit 1
+}
+
+#####
+
 POLL_INTERVAL=1 # sec
 
 #####
 
 typeset COORD_POINT
-typeset -a DATAMON_CMDS
 typeset BUNDLE_ID_FILE
 
 SLEEP_INSTEAD_OF_EXIT=
 
-while getopts sc:d:i: opt; do
-    case $opt in
-        (s)
-            SLEEP_INSTEAD_OF_EXIT=true
-            ;;
-        (c)
-            COORD_POINT="$OPTARG"
-            ;;
-        (d)
-            datamon_cmd=$(echo "$OPTARG" |tr '\t' ' ' |tr -s ' ' | \
-                              sed 's/^ //' |sed 's/ $//')
-            DATAMON_CMDS=($datamon_cmd $DATAMON_CMDS)
-            ;;
-        (i)
-            BUNDLE_ID_FILE="$OPTARG"
-            ;;
-        (\?)
-            echo "Bad option, aborting."
-            return 1
-            ;;
-    esac
+####
+
+# deserialize an associate array from a scalar
+# example usage
+# typeset -A opts_global_dict
+# opts_global_dict=($(deserialize_dict $opts_global))
+# todo: implement via set variables by reference as in
+# deserialize_dict $opts_global opts_global_dict
+#   .. access by reference is via ${(P)var_name}.
+#   unsure how to set.
+deserialize_dict() {
```

Comment:
All of this can be avoided if we have a set of known environment variables.
$DM_REPO
$DM_BUNDLE_LABLE
$DM_FUSE_STREAM

Reply:
@galvare2 would you prefer to generate environment variables as dictionaries or have independent environment variables?

Reply:
It's unimportant to me, but I will say that this serialization scheme seems a bit complicated, so whichever implementation reduces the complexity is fine with me. Perhaps it would be preferable to use a standard, well-known serialization such as json or similar?

Reply:
to be an internal implementation detail occluded by pending golang mentioned in [the comment](https://github.com/oneconcern/datamon/pull/270#issuecomment-543291247) following this morning's discussion